src/solver/ia_main.jl

```diff
@@ -0,0 +1,246 @@
+using Symbolics
+
+function isolate(lhs, var)
+    rhs = Vector{Any}([0])
+    lhs = unwrap(lhs)
+    while !isequal(lhs, var)
+        subs, poly = filter_poly(lhs, var)
+        try
+            if check_poly_inunivar(poly, var)
```

Comment:
It seems that if `lhs` is not supported, this goes into infinite loop.
Instead of this, you can check if `lhs` stabilizes, and, if so, throw an error that we do not support it.